tests/fetch_embedding.py

```diff
@@ -1,27 +1,29 @@
-
-import openai
-import os 
+import os
 import sys
-import dotenv
-dotenv.load_dotenv()
 
-openai.api_type = "azure"
-openai.api_key = os.environ["OPENAI_API_KEY"]
-openai.api_base = f"https://{os.environ['AZURE_OPENAI_SERVICE']}.openai.azure.com"
-openai.api_version = "2023-05-15"
+from dotenv import load_dotenv
+from openai import AzureOpenAI
+
+load_dotenv()
+
+
+client = AzureOpenAI(
+    api_key=os.environ["OPENAI_API_KEY"],
+    azure_endpoint=f"https://{os.environ['AZURE_OPENAI_SERVICE']}.openai.azure.com",
```

Comment:
This is a personal preference, but I find defining environment has a constant very helpful for change management. Here's an example:

```python
OPENAI_API_KEY = 'OPENAI_API_KEY'
AZURE_OPENAI_SERVICE = 'AZURE_OPENAI_SERVICE'
```

Reply:
I agree. I would suggest a good round of refactoring on this codebase, especially since the delay issues have been fixed to a tolerable amount.